### PR TITLE
Add command to allow adding/changing/removing server password

### LINKED ISSUE #125: Ability to change/add/remove server password at runtime
PF had an `sv_pass` command, allowing someone at the server console or using rcon to change the server password (or remove it, if running `sv_pass` with no argument). Would be great if Dash had this as well.

### PR DESCRIPTION
**Pull request by GooberRF**
This PR adds the `server_password` command, which can be used either as server or via rcon to add a password to a server without one, change the existing password to something else, or remove the password entirely.

Syntax is `server_password <password>`

Resolves #125

### CODE REVIEW

File: game_patch/os/commands.cpp [hunk 1]
```diff
@@ -92,6 +92,27 @@ DcCommandAlias map_info_cmd{
...
+    "server_password",
+    [](std::optional<std::string> pattern) {
+        if (!rf::is_multi || !rf::is_server) {
+            rf::console::print("This command can only be run as a server!");
+            return;
+        }            
+
+        if (pattern) {
+            rf::netgame.password = pattern.value().c_str();
+            rf::console::print("Server password set to: {}", *pattern);
```
Comment (is-this-c) [inline game_patch/os/commands.cpp:105]:
```suggestion
            rf::console::print("Server password set to: {}", rf::netgame.password);
```

File: game_patch/os/commands.cpp [hunk 2]
```diff
@@ -92,6 +92,27 @@ DcCommandAlias map_info_cmd{
     level_info_cmd,
 };
 
+ConsoleCommand2 server_password_cmd{
+    "server_password",
+    [](std::optional<std::string> pattern) {
```
Comment (is-this-c) [inline game_patch/os/commands.cpp:97]:
```suggestion
    [](std::optional<std::string> pass) {
```

**Comment by is-this-c** [comment]
I think `sv_pass` is better. It is simpler and was previously used by PF.

**Comment by GooberRF** [comment]
> I think `sv_pass` is better. It is simpler and was previously used by PF.

I am personally of the view that adjusting the naming convention for all console commands to prioritize short length and `sv_`, `cl_`, `r_`, `mp_`, etc. prefixes (like id tech games) would be ideal, but unless rafalh were to decide to go that way, it's best to continue with the approach already established rather than having some commands follow the existing convention and some follow the id tech/PF approach.

`sv_pass` in my view would be a better way to name this command, but it doesn't match the established convention unlike `server_password`.

**Comment by is-this-c** [comment]
Don't those other games have tons of console naming inconsistencies anyway?

**Comment by GooberRF** [comment]
> Don't those other games have tons of console naming inconsistencies anyway?

Not generally, but even if they did, I don't think Dash should. If a standard is established, it should be maintained unless there's a very good reason not to (or if the standard changes).

Again, I think using prefixes like this would be good, but it explicitly goes against the established standard, so I have no intention to use that approach unless the project owner wants to change the established standard.

**Review by rafalh — APPROVED** [review]

**Pull request merged**